### ISSUE TITLE
Node crash with ERR_STREAM_DESTROYED when rotating logs

### ISSUE THREAD

**Issue by ChrisEdgington** [winstonjs/winston-daily-rotate-file#162]
I was using a simple file transport in my server. 

```
       new winston.transports.File({ filename: C.DEFAULT_LOG_FILENAME })
```

That has been working fine throughout development. Now that I'm going to production, I added winston-daily-rotate-file, so I replaced the File transport listed above with this:

```
      new winston.transports.DailyRotateFile({ 
          filename      : C.DEFAULT_LOG_FILENAME + '-%DATE%.log',
          dirname       : './logs',
          datePattern   : 'YYYY-MM-DD:HH',
          zippedArchive : false,
          maxSize       : '2m',
          maxFiles      : '14d',
        })
```

But I found that my node server was crashing. Turns out it was crashing when it hit the file size limit and rotated log files. I changed the maxSize to "4k" and was able to get it to crash easily by generating log data. Here is the message from the crash:

```
Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed
    at doWrite (_stream_writable.js:406:19)
    at clearBuffer (_stream_writable.js:517:5)
    at onwrite (_stream_writable.js:465:7)
    at fs.write (fs.js:2282:5)
    at FSReqWrap.wrapper [as oncomplete] (fs.js:643:5)
```

I'm running node version is 10.3.0 on macOS 10.13.5. I'm new to node / winston, so its not clear to me if this is my problem, a problem in node, or a problem in winston.

**Comment by mattberther** [comment]
@ChrisEdgington Can you please try using https://github.com/winstonjs/winston-daily-rotate-file/pull/156 and advise if your issue is resolved?

Thanks!

**Comment by mattberther** [comment]
Merged #156 and published to npm as `winston-daily-rotate-file@3.3.1`

**Comment by hpsharon** [comment]
Happens to me in version 3.9.0

```
Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed
    at doWrite (_stream_writable.js:413:19)
    at writeOrBuffer (_stream_writable.js:401:5)
    at WriteStream.Writable.write (_stream_writable.js:301:11)
```

**Comment by Nesher123** [comment]
Same here, I think that the issue happens since my project uses pm2 in order to run the app.
Is there a way to make it work without disabling pm2 completely? 

Thank you

**Comment by mattberther** [comment]
@hpsharon @Nesher123 To troubleshoot this, could you provide the smallest bit of code that consistently reproduces the error please?

**Comment by Nesher123** [comment]
Hey @mattberther,
I'm attaching the full error 

`Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed
    at doWrite (_stream_writable.js:413:19)
    at writeOrBuffer (_stream_writable.js:401:5)
    at WriteStream.Writable.write (_stream_writable.js:301:11)
    at Object.<anonymous> (C:\Program Files\nesher\next\someFolder\NodeServer\node_modules\file-stream-rotator\FileStreamRotator.js:478:26)
    at DailyRotateFile.log (C:\Program Files\nesher\next\someFolder\NodeServer\node_modules\winston-daily-rotate-file\daily-rotate-file.js:146:24)
    at DailyRotateFile._write (C:\Program Files\nesher\next\someFolder\NodeServer\node_modules\winston-transport\index.js:102:17)
    at doWrite (C:\Program Files\nesher\next\someFolder\NodeServer\node_modules\readable-stream\lib\_stream_writable.js:428:64)
    at writeOrBuffer (C:\Program Files\nesher\next\someFolder\NodeServer\node_modules\readable-stream\lib\_stream_writable.js:417:5)
    at DailyRotateFile.Writable.write (C:\Program Files\nesher\next\someFolder\NodeServer\node_modules\readable-stream\lib\_stream_writable.js:334:11)
    at DerivedLogger.ondata (C:\Program Files\nesher\next\someFolder\NodeServer\node_modules\winston\node_modules\readable-stream\lib\_stream_readable.js:662:20)
    at DerivedLogger.emit (events.js:201:15)
    at DerivedLogger.EventEmitter.emit (domain.js:471:20)
    at addChunk (C:\Program Files\nesher\next\someFolder\NodeServer\node_modules\winston\node_modules\readable-stream\lib\_stream_readable.js:297:12)
    at readableAddChunk (C:\Program Files\nesher\next\someFolder\NodeServer\node_modules\winston\node_modules\readable-stream\lib\_stream_readable.js:279:11)
    at DerivedLogger.Readable.push (C:\Program Files\nesher\next\someFolder\NodeServer\node_modules\winston\node_modules\readable-stream\lib\_stream_readable.js:240:10)
    at DerivedLogger.Transform.push (C:\Program Files\nesher\next\someFolder\NodeServer\node_modules\winston\node_modules\readable-stream\lib\_stream_transform.js:139:32)`

And the logger file here:
[Logger.txt](https://github.com/winstonjs/winston-daily-rotate-file/files/3244575/Logger.txt)

And finally, `app.use(logger)` after importing it.
I'm using pm2 to run the app.

Thanks

**Comment by mattberther** [comment]
I'm moving this issue to #226, which more accurately describes the problem (`winston-daily-rotate-file` does not work well with process managers).

**Comment by shirakaba** [comment]
To anyone it may help: I got this error when I ran my app initially with `sudo` (meaning that logs would be written as `root:root`), then again without `sudo` (meaning that logs would be written as `ubuntu:ubuntu`, and thus lacked privileges to rotate the logs).

So in my case, it wasn't a fault of Winston. Just a standard consequence of permissions management. To avoid it, always run the app as a user in the same usergroup as you intend to run it with in future.
